apidef/migration.go

```diff
@@ -78,3 +78,37 @@ func (a *APIDefinition) MigrateVersioning() (versions []APIDefinition, err error
 
 	return
 }
+
+func (a *APIDefinition) MigrateCachePlugin() (err error) {
+	if a.VersionDefinition.Enabled || len(a.VersionDefinition.Versions) != 0 {
+		return errors.New("not migratable - new versioning is enabled")
+	}
+
+	if a.VersionData.NotVersioned && len(a.VersionData.Versions) > 1 {
+		return errors.New("not migratable - if not versioned, there should be 1 version info in versions map")
+	}
+
+	for vName, vInfo := range a.VersionData.Versions {
```

Comment:
We don't need to iterate over versions. As we will call this function after migrating versioning, every definition will have just one vInfo like `a.VersionData.Versions[""]`. See https://github.com/TykTechnologies/tyk/pull/3805/files#diff-e1d9b55a26f9d6225d56d6f0161959217308e5ad4d6934e7d7df4595d9c2a130R104